engine/flink/api/src/main/scala/pl/touk/nussknacker/engine/flink/api/process/FlinkSource.scala

```diff
@@ -1,58 +1,125 @@
 package pl.touk.nussknacker.engine.flink.api.process
 
+import org.apache.flink.api.common.functions.MapFunction
 import org.apache.flink.api.common.typeinfo.TypeInformation
 import org.apache.flink.streaming.api.TimeCharacteristic
 import org.apache.flink.streaming.api.functions.source.SourceFunction
 import org.apache.flink.streaming.api.scala.{DataStream, StreamExecutionEnvironment}
-import pl.touk.nussknacker.engine.api.MethodToInvoke
-import pl.touk.nussknacker.engine.api.process.{Source, SourceFactory}
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.{Context, MethodToInvoke}
+import pl.touk.nussknacker.engine.api.process.{Source, SourceFactory, SourceTestSupport}
+import pl.touk.nussknacker.engine.api.typed.typing
 import pl.touk.nussknacker.engine.flink.api.compat.ExplicitUidInOperatorsSupport
 import pl.touk.nussknacker.engine.flink.api.timestampwatermark.TimestampWatermarkHandler
 
 import scala.reflect._
 
 /**
   * Source with methods specific for Flink
-  * @tparam T - type of event that is generated by this source. This is needed to handle e.g. syntax suggestions in UI
+  *
+  * @tparam Raw - type of raw event that is generated by flink source function.
   */
-trait FlinkSource[T] extends Source[T] {
+//TODO: remove type T parameter
+trait FlinkSource[Raw] extends Source[Raw] {
```

Comment:
I think that here shouldn't be Raw - Raw is only in case of `FlinkIntermediateRawSource` / `FlinkContextInitializer`. In simple, basic case this `T` is final output variable type

Reply:
Good point. Reverted to `FlinkSource[T]`.